h/api/events.py

```diff
@@ -1,8 +1,33 @@
 # -*- coding: utf-8 -*-
-class AnnotationEvent(object):
+
+# Types representing events stored in the notification queue
+# (see queue.py)
+
+class Event(object):
+    def __init__(self, request):
+        self.request = request
+
+class AnnotationEvent(Event):
     """An event representing an action on an annotation."""
 
     def __init__(self, request, annotation, action):
-        self.request = request
+        Event.__init__(self, request)
         self.annotation = annotation
         self.action = action
+
+class UserStatusEvent(Event):
```

Comment:
Hopefully you'll be glad to hear that very little of this is actually needed. Sorry if I didn't get the message across clearly enough -- you asked in Slack and I tried to explain that you didn't need to follow the `AnnotationEvent` pattern.

So, I'd revert all changes to this file and `h/api/queue.py`, and just push a JSON blob onto the queue in the group view functions. I'll annotate those below.